MagazineLayout/LayoutCore/SectionModel.swift

```diff
@@ -341,108 +327,81 @@ struct SectionModel {
       size: itemModel.size)
   }
 
-  private mutating func recomputeItemPositionsIfNecessary() {
-    guard let startingIndex = indexOfFirstInvalidatedItem else { return }
-    indexOfFirstInvalidatedItem = nil
-
-    guard startingIndex >= 0 && startingIndex <= itemModels.count else {
-      assertionFailure("Invalid `startingIndex` / `indexOfFirstInvalidatedItem` (\(startingIndex)).")
+  private mutating func calculateElementFramesIfNecessary() {
+    guard var rowIndex = indexOfFirstInvalidatedRow else { return }
+    guard rowIndex >= 0 else {
```

Comment:
Would this also be invalid if numberOfItems == 0 and rowIndex == 0?

Reply:
Not necessarily, since there could be as many as 2 valid row indices without there being any items in the section.

```
0: [Header]
```

```
0: [Footer]
```

```
0: [Header]
1: [Footer]
```

Really, this is just a guard against receiving a negative row index, which would definitely be a programmer error (on our part).